test/ametsuchi/index/merkle_test.cc

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ametsuchi/index/merkle.h>
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <numeric>
+#include <cmath>
+#include <vector>
+#include <iostream>
+
+namespace ametsuchi {
+namespace index {
+namespace merkle {
+
+constexpr auto size = 10;
+
+TEST(Merkle, Creation) {
+  NarrowMerkleTree<uint64_t> tree([](auto i, auto j) { return i + j; });
+}
+
+TEST(Merkle, Addition) {
+  NarrowMerkleTree<uint64_t> tree([](auto i, auto j) { return i + j; });
+  /*
+   *                              55
+   *              28                           27
+   *       6              22              27
+   *   1       5       9      13      17       10
+   * 0   1   2   3   4   5   6   7   8   9   10
+   */
+  for (size_t i = 0, sum = 0; i < size * size; ++i, sum += i) {
```

Comment:
Nope, it's fine. Just wanted to check for quite bigger amount rather than "tested by hands"